network/stream/peer.go

```diff
@@ -192,6 +194,8 @@ func (p *Peer) sealWant(w *want) error {
 	}
 	p.mtx.Lock()
 	delete(p.openWants, w.ruid)
+	s := fmt.Sprintf("%s_%t", w.stream.String(), w.head)
```

Comment:
Consider creating one function for constructing the key as it is used twice even it different files.